apps/gdalalg_raster_index.h

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster polygonize" subcommand
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#ifndef GDALALG_RASTER_INDEX_INCLUDED
+#define GDALALG_RASTER_INDEX_INCLUDED
+
+#include "gdalalg_vector_output_abstract.h"
+
+//! @cond Doxygen_Suppress
+
+/************************************************************************/
+/*                     GDALRasterIndexAlgorithm                         */
+/************************************************************************/
+
+class CPL_DLL GDALRasterIndexAlgorithm /* non final */
+    : public GDALVectorOutputAbstractAlgorithm
+{
+  public:
+    static constexpr const char *NAME = "index";
+    static constexpr const char *DESCRIPTION =
+        "Create a vector index of raster datasets.";
+    static constexpr const char *HELP_URL = "/programs/gdal_raster_index.html";
+
+    GDALRasterIndexAlgorithm();
+
+    GDALRasterIndexAlgorithm(const std::string &name,
+                             const std::string &description,
+                             const std::string &helpURL);
+
+  protected:
+    void AddCommonOptions();
+
+    // Virtual method that may be overridden by derived classes to add options
+    // to GDALTileIndex()
+    virtual bool AddExtraOptions([[maybe_unused]] CPLStringList &aosOptions)
+    {
+        return true;
+    }
+
+    std::vector<GDALArgDatasetValue> m_inputDatasets{};
+
+  private:
+    bool RunImpl(GDALProgressFunc pfnProgress, void *pProgressData) override;
+
+    bool m_recursive = false;
+    std::vector<std::string> m_filenameFilter{};
+    double m_minPixelSize = 0;
+    double m_maxPixelSize = 0;
```

Comment:
I think that minPixelSize and maxPixelSize will always be changed into some bigger values, right?

Reply:
> I think that minPixelSize and maxPixelSize will always be changed into some bigger values, right?

yes those are initialization values that are not valid as values you can set